kraken-build/src/kraken/std/cargo/tasks/cargo_publish_task.py

```diff
@@ -37,6 +41,36 @@ class CargoPublishTask(CargoBuildTask):
     #: Cargo.toml which to temporarily bump
     cargo_toml_file: Property[Path] = Property.default("Config.toml")
 
+    #: Allow Overwrite of existing packages
+    allow_overwrite: Property[bool] = Property.default(False)
+
+    def prepare(self) -> TaskStatus | None:
+        """Checks if the crate@version already exists in the registry. If so, the task will be skipped"""
+        if self.allow_overwrite.get():
+            return TaskStatus.pending()
+
+        package_name = self.package_name.get()
```

Comment:
Tbh. I'm not sure why we have it already as a property when it's not referenced in the implementation of `execute()` before.. 🤔

Reply:
[cargo_publish_task.py#L86-L88](https://github.com/kraken-build/kraken/blob/86d17ed/kraken-build/src/kraken/std/cargo/tasks/cargo_publish_task.py#L86-L88) hints that it can be None. I guess in that case, we should pick the package_name from the `cargo.toml` file

Reply:
After digging a bit deeper into the usage of `package_name` in our downstream internal Kraken module I found:

```py
        for member in self.metadata.workspaceMembers:
            if member.name not in self.exclude:
                members_publish_task[member.name] = cargo_publish(
                    publish_repo,
                    incremental=False,
                    verify=not is_playground and self.verify,
                    retry_attempts=5,
                    package_name=member.name,
                    project=self.project,
                    version=version,
                    cargo_toml_file=member.manifest_path,
                )
```

Where `metadata` is a `CargoMetadata` which is generated with `CargoMetadata.read()`, and ends up deriving the `member.name` here:

https://github.com/kraken-build/kraken/blob/62cd7f859d350a8c5616e223c460ad66afb0acd5/kraken-build/src/kraken/std/cargo/manifest.py#L127-L133

So I think we're good here, `package_name` is populated with the actual Crate name and not the [workspace member name](https://doc.rust-lang.org/cargo/reference/workspaces.html#the-members-and-exclude-fields).